napari_brightness_contrast/_dock_widget.py

```diff
@@ -177,25 +197,28 @@ def redraw(self, rebuild_gui=True):
                 layout.itemAt(i).widget().setParent(None)
             for i, layer in enumerate(self.selected_image_layers()):
 
-                row = LayerContrastLimitsWidget(layer, colors[i], all_minimum, all_maximum, self)
+                row = LayerContrastLimitsWidget(
+                    layer, colors[i], all_minimum, all_maximum, self
+                )
 
                 layout.addWidget(row)
 
         # patch events
         selected_layers = self.selected_image_layers()
         for layer in self.viewer.layers:
-            layer.events.contrast_limits.disconnect(self._data_changed_event)
-            layer.events.data.disconnect(self._data_changed_event)
-            if layer in selected_layers:
-                layer.events.contrast_limits.connect(self._data_changed_event)
-                layer.events.data.connect(self._data_changed_event)
+            if hasattr(layer.events, "contrast_limits"):
```

Comment:
Here, this is the key line you suggested, to check that "contrast limits" can be changed.

Reply:
ok cool, thanks!